### PR TITLE
Additional Bundling of Core Packages

### LINKED ISSUE #347: Uncaught errors still show Atom instead of Pulsar
### Thanks in advance for your bug report!

- [ ] Have you reproduced issue in [safe mode](https://pulsar-edit.dev/docs/atom-archive/faq/sections/what-does-safe-mode-do.html)?
- [ ] Have you used the [debugging guide](https://pulsar-edit.dev/docs/atom-archive/hacking-atom/#debugging) to try to resolve the issue?
- [X] Have you checked our [FAQs](https://pulsar-edit.dev/docs/launch-manual/sections/faq/) to make sure your question isn't answered there?
- [X] Have you checked to make sure your issue does not already [exist](https://github.com/issues?q=is%3Aissue+user%3Apulsar-edit+is%3Aopen)?
- [ ] Have you checked you are on the latest release of Pulsar?

### What happened?

Uncaught error message still shows all Atom branding, and possibly linked to the Atom repo (sorry I didn't think to check, and I've since closed it, if I get a new one I'll update this issue).  The thing that caused the error is [#134](https://github.com/pulsar-edit/pulsar/issues/134).

error message image

### Pulsar version

1.100.0-beta

### Which OS does this happen on?

🍎 macOS

### OS details

macOS 13

### Which CPU architecture are you running this on?

64-bit(x86_64)

### What steps are needed to reproduce this?

Do something that will produce an uncaught error (for we it was a "Cannot decorate a destroyed marker").

### Additional Information:

_No response_

### PR DESCRIPTION
**Pull request by confused-Techie**
This PR is just like the many others I've done. This should result in zero changes of how Pulsar functions or behaves, just bundles more of our already core packages into the core repo.

Packages Bundled in this PR:
* `autosave`
* ~~`notifications`~~
* `whitespace` (This one was bundled but had never been locally installed)
* `bracket-matcher`
* `timecop`
* `keybinding-resolver`

As expected each bundle has also updated the `README.md` and `package.json` as needed. 
These bundles will also bring Pulsar these new PRs:

* [`autosave#1 - Cleanup and rename @Sertonix`](https://github.com/pulsar-edit/autosave/pull/1)
* [`autosave#2 - removed fs-plus dependency @Sertonix `](https://github.com/pulsar-edit/autosave/pull/2)
* ~~[`notifications#1 - Cleanup and rename @Sertonix `](https://github.com/pulsar-edit/notifications/pull/1)~~
* ~~[`notifcations#2 - reject promise with Error instance @Sertonix `](https://github.com/pulsar-edit/notifications/pull/2)~~
* ~~[`notifications#3 - Add our Testing Action @confused-Techie `](https://github.com/pulsar-edit/notifications/pull/3)~~
* ~~[`notifications#4 - Change atom strings to pulsar @mdibella-dev `](https://github.com/pulsar-edit/notifications/pull/4)~~
* ~~[`notifications#5 - Bump to v3.2 of action-pulsar-dependency @confused-Techie `](https://github.com/pulsar-edit/notifications/pull/5)~~
* ~~[`notifications#6 - Fix all Tests @confused-Techie `](https://github.com/pulsar-edit/notifications/pull/6)~~
* [`bracket-matcher#1 - Rename A[a]tom -> P[p]ulsar @Spiker985 `](https://github.com/pulsar-edit/bracket-matcher/pull/1)
* [`bracket-matcher#2 - cleanup .md and rename repo url @Sertonix `](https://github.com/pulsar-edit/bracket-matcher/pull/2)
* [`bracket-matcher#3 - Fixing tests that need to run locally @mauricioszabo `](https://github.com/pulsar-edit/bracket-matcher/pull/3)
* [`timecop#1 - cleanup and rename @Sertonix `](https://github.com/pulsar-edit/timecop/pull/1)
* [`keybinding-resolver#1 - cleanup and rename @Sertonix `](https://github.com/pulsar-edit/keybinding-resolver/pull/1)

By bringing in these updates we will also be able to resolve some issues:

~~Resolves # 347~~

EDIT: The tests on `notifications` seem to rely on the fact that it's a separate package, and bundling did not seem possible until that is updated. So I've removed the changes we would get by including it, and instead will handle bumping it in a standalone PR

### CODE REVIEW

File: packages/README.md [hunk 1]
```diff
@@ -19,12 +19,12 @@ See [RFC 003](https://github.com/atom/atom/blob/master/docs/rfcs/003-consolidate
 | **autocomplete-plus** | [`./autocomplete-plus`][./autocomplete-plus] |  |
 | **autocomplete-snippets** | [`./autocomplete-snippets`](./autocomplete-snippets) |  |
 | **autoflow** | [`./autoflow`](./autoflow) | |
-| **autosave** | [`pulsar-edit/autosave`][autosave] | [#17834](https://github.com/atom/atom/issues/17834) |
+| **autosave** | [`./autosave`][./autosave] | |
```
Comment (Daeraxa) [inline packages/README.md:22]:
```suggestion
| **autosave** | [`./autosave`](./autosave) | |
```

File: packages/README.md [hunk 2]
```diff
@@ -19,12 +19,12 @@ See [RFC 003](https://github.com/atom/atom/blob/master/docs/rfcs/003-consolidate
...
 | **background-tips** | [`./background-tips`](./background-tips) | |
 | **base16-tomorrow-dark-theme** | [`./base16-tomorrow-dark-theme`](./base16-tomorrow-dark-theme) | |
 | **base16-tomorrow-light-theme** | [`./base16-tomorrow-light-theme`](./base16-tomorrow-light-theme) | |
 | **bookmarks** | [`./bookmarks`](./bookmarks) | |
-| **bracket-matcher** | [`atom/bracket-matcher`][bracket-matcher] |  |
+| **bracket-matcher** | [`./bracket-matcher`][./bracket-matcher] |  |
```
Comment (Daeraxa) [inline packages/README.md:27]:
```suggestion
| **bracket-matcher** | [`./bracket-matcher`](./bracket-matcher) |  |
```

File: packages/README.md [hunk 3]
```diff
@@ -39,7 +39,7 @@ See [RFC 003](https://github.com/atom/atom/blob/master/docs/rfcs/003-consolidate
 | **grammar-selector** | [`./grammar-selector`](./grammar-selector) | |
 | **image-view** | [`./image-view`](./image-view) | |
 | **incompatible-packages** | [`./incompatible-packages`](./incompatible-packages) | |
-| **keybinding-resolver** | [`atom/keybinding-resolver`][keybinding-resolver] | [#18275](https://github.com/atom/atom/issues/18275) |
+| **keybinding-resolver** | [`./keybinding-resolver`][./keybinding-resolver] | |
```
Comment (Daeraxa) [inline packages/README.md:42]:
```suggestion
| **keybinding-resolver** | [`./keybinding-resolver`](./keybinding-resolver) | |
```

File: packages/README.md [hunk 4]
```diff
@@ -77,7 +77,7 @@ See [RFC 003](https://github.com/atom/atom/blob/master/docs/rfcs/003-consolidate
 | **line-ending-selector** | [`./line-ending-selector`](./line-ending-selector) | |
 | **link** | [`./link`](./link) | |
 | **markdown-preview** | [`./markdown-preview`][./markdown-preview] |  |
-| **notifications** | [`atom/notifications`][notifications] | [#18277](https://github.com/atom/atom/issues/18277) |
+| **notifications** | [`./notifications`][./notifications] | |
```
Comment (Daeraxa) [inline packages/README.md:80]:
```suggestion
| **notifications** | [`./notifications`](./notifications) | |
```

File: packages/README.md [hunk 5]
```diff
@@ -93,23 +93,18 @@ See [RFC 003](https://github.com/atom/atom/blob/master/docs/rfcs/003-consolidate
 | **styleguide** | [`./styleguide`][./styleguide] | |
 | **symbols-view** | [`pulsar-edit/symbols-view`][symbols-view] |  |
 | **tabs** | [`./tabs`](./tabs) |  |
-| **timecop** | [`pulsar-edit/timecop`][timecop] | [#18272](https://github.com/atom/atom/issues/18272) |
+| **timecop** | [`./timecop`][./timecop] | |
```
Comment (Daeraxa) [inline packages/README.md:96]:
```suggestion
| **timecop** | [`./timecop`](./timecop) | |
```

**Review by Daeraxa — COMMENTED** [review]
The second brackets just needed to be changed from a reference link as it was before to a normal link

**Comment by confused-Techie** [comment]
> The second brackets just needed to be changed from a reference link as it was before to a normal link

Thanks for catching that, I've committed your suggestions!

**Comment by confused-Techie** [comment]
So a quick update, the Windows Cirrus CI test failed, but I am currently rerunning that since it only timed out.

Otherwise here's a look of our pass fail rate:

* find-and-replace: 52 Failures - Within Expected Failure Rate ✅
* notifications: 15 Failures - 0 Expected Failures ❌
* symbols-view: 2 Failures - Within Expected Failure Rate ✅
* tree-view: 2 Failures - Within Expected Failure Rate ✅

So I'll go ahead and see what I can find out about the tree-view tests. Just weird since they were passing on the tree view repo itself. But I'll take a further look at them.

**Comment by DeeDeeG** [comment]
I suppose we should add any newly bundled packages to the packages tests CI job.

--> https://github.com/pulsar-edit/pulsar/blob/d8b0a3381aec5814205dc433736a47b95ddfe149/.github/workflows/package-tests-linux.yml#L49-L58

Since (perhaps surprisingly) they are all hard-coded, rather than like programatically `cd`ing into there and listing out all the subdirectories into the test matrix or something like that.

**Comment by confused-Techie** [comment]
@DeeDeeG So my understanding, is that we tell Pulsar to test the core package of a certain name and it tests whatever it's using. Whether installed from the `package.json` or included in the core repos. And I'm pretty sure all of them are already listed there. I'll go through and double check, but my understanding is that the existing tests will test all packages included here already, as we have before just installed from a different source

**Comment by DeeDeeG** [comment]
Oh blarg, you're right. Nevermind, then!

_(Off-topic tangent: So that's why we don't just `cd` into the package dir and then install only its own dependencies before testing... Several of the packages wouldn't exist on disk to test, in that case!)_

**Comment by DeeDeeG** [comment]
Continuing the off-topic tangent, we should really cache the main `node_modules` and `packages/` dirs from the `build editor` job, and re-use said cached dirs for the individual packages' test jobs, it'd speed them all up _tremendously_.

**Comment by DeeDeeG** [comment]
So, for the `notifications` package tests, we're getting a number of these:

```
Invalid Version: file:./packages/notifications
TypeError: Invalid Version: file:./packages/notifications
    at new SemVer (/home/runner/work/pulsar/pulsar/node_modules/notifications/node_modules/semver/semver.js:295:11)
    at SemVer.compare (/home/runner/work/pulsar/pulsar/node_modules/notifications/node_modules/semver/semver.js:348:13)
    at compare (/home/runner/work/pulsar/pulsar/node_modules/notifications/node_modules/semver/semver.js:571:31)
    at Function.gte (/home/runner/work/pulsar/pulsar/node_modules/notifications/node_modules/semver/semver.js:620:10)
    at /home/runner/work/pulsar/pulsar/node_modules/notifications/lib/user-utilities.coffee:148:49
    at async Promise.all (index 2)
```

Some code in there is unhappy about trying to use `file:./packages/notifications` as a semver thing.

**Comment by confused-Techie** [comment]
Good catch there @DeeDeeG, looking at that code yeah it looks like it's grabbing it's exact version from the `packageDependencies` themselves. Weird it's only failing now, considering it's always been encountering these items. But suppose maybe it only checks for it's own version there. Might have to think about how it's best to fix that one

**Comment by confused-Techie** [comment]
So now with me removing `notifications` from this PR here are our tests:
* `find-and-replace`: 56 Failures ✅
* `settings-view`: 1 Failure ❌ (But this is a repo wide issue, being discussed on discord, and means nothing for this specific PR)
* `symbols-view`: 2 Failures ✅
* `tree-view`: 2 Failures ✅

So this should be good to merge unless anyone has any concerns!

**Comment by confused-Techie** [comment]
Alright, had to merge from main to resolve a conflict, which brought over the broken status of our CI.

Just updated it again, now that it's all over with. Will let tests complete one last time, and if things match what I'm seeing over on #469 then I'll feel confident this doesn't cause any issues and merge it.

**Pull request merged**